assets/js/translator.js

```diff
@@ -1111,14 +1114,70 @@ function setRecentDstLangs(langs /*: string[] */) {
     return langs;
 }
 
+function dictionaryLookup() {
+    if(modeEnabled('lookup')) {
+        if(currentLookupRequest) {
+            currentLookupRequest.abort();
+        }
+
+        $('#dictionaryLookupResult').empty();
+        var wordToLookup = $('#originalText').val().trim().split(' '); // eslint-disable-line newline-per-chained-call
+        if(wordToLookup.length <= 3 && wordToLookup[0] !== '') {
+            currentLookupRequest = callApy({
+                data: {
+                    'langpair': curSrcLang + '|' + curDstLang,
+                    'q': $('#originalText').val()
+                },
+                success: handleDictionaryLookupSuccessResponse,
+                error: handleDictionaryLookupErrorResponse,
+                complete: function () {
+                    ajaxComplete();
+                    currentLookupRequest = undefined;
+                }
+            }, '/dictionaryLookup');
+        }
+    }
+
+    function handleDictionaryLookupSuccessResponse(data) {
+        $('#dictionaryLookup').removeClass('hide');
+        if($('#currentWordForLookup').length === 0) {
+            var currentWord = $('<span id="currentWordForLookup">');
+            currentWord.text($('#originalText').val());
+            $('#dictTranslation').text($('#dictTranslation').text().replace('{{word}}', ''));
+            $('#dictTranslation').append(currentWord);
+        }
+        else {
+            $('#currentWordForLookup').text($('#originalText').val());
+        }
+        for(var analysisType in data) {
+            var values = $('<ul id="dictionaryTranslationContents">');
+            for(var translationContent = 0; translationContent < data[analysisType].length; translationContent++) {
+                var livalues = $('<li>');
+                livalues.text(data[analysisType][translationContent].replace('#', ''));
+                values.append(livalues);
+            }
+            var dlSpan = $('<span id="dictDLHeaders">');
+            dlSpan.attr({'data-text': 'POS_' + analysisType});
+            dlSpan.text(dynamicLocalizations['fallback']['POS_' + analysisType]);
```

Comment:
@sushain97 @shardulc Right now the problem is that the text that is suppose to be inside the because of the data-text isn't rendering. It is probably because all the data-text stuff gets rendered in the beginning of the website load. Is there a function I can call to render a specific data-text or re-render everything? Any hints or help is appreciated.

The dlSpan.text(dynamicLocalizations['fallback']['POS_' + analysisType]); is currently placeholder code before I can find an actual fix.

I tried looking through this repository for things related to data-text, and I found this: https://github.com/goavki/apertium-html-tools/blob/780e615ac7a10f51caffd9e2dbc6a28e3e3f1d8f/tools/localise-html.py, though I don't know if it is useful in my case.

Reply:
Found a similar usage here: https://github.com/goavki/apertium-html-tools/blob/97f233289327ae292db76ebd29d959a06bd6f157/assets/js/translator.js#L840

What is funny is that the data-text tag doesn't do anything. It is actually the line above (line 839) that makes the text appear.

Reply:
Yeah, there's no need to do the `data-text` thing.